### PR TITLE
API Review - Microsoft.AspNetCore.Testing

### PR DESCRIPTION
**Pull request by rafal-mz**
The package contains helpers on top of AspNetCore that are meant to help shorten testing code and provide common helpers. All APIs can be considered as experimental, though in 1P they were stable for last few months.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/extensions/pull/4468)

### CODE REVIEW

File: API/WithDocs/Microsoft.AspNetCore.Testing/ServiceFakesExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Hosting.Server;
+using Microsoft.Extensions.Hosting;
+
+namespace Microsoft.AspNetCore.Testing;
+
+/// <summary>
+/// Extension methods supporting Kestrel server unit testing scenarios.
+/// </summary>
+public static class ServiceFakesExtensions
```
Comment (geeknoid) [inline API/WithDocs/Microsoft.AspNetCore.Testing/ServiceFakesExtensions.cs:16]: Can anybody come up with a better name here?

**Comment by joperezr** [comment]
@rafal-mz @geeknoid is this still relevant? This package is already being reviewed and we are adding a readme [here](https://github.com/dotnet/extensions/pull/4561), so we were not aware of any pending changes here. cc: @Tratcher

**Comment by joperezr** [comment]
Marking as No Merge as this is API Review only.

File: API/WithDocs/Microsoft.AspNetCore.Testing/ServiceFakesExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    public static IWebHostBuilder ListenHttpsOnAnyPort(this IWebHostBuilder builder, X509Certificate2? sslCertificate = null);
+
+    /// <summary>
+    /// Creates an <see cref="T:System.Net.Http.HttpClient" /> to call the hosted application.
+    /// </summary>
+    /// <param name="host">An <see cref="T:Microsoft.Extensions.Hosting.IHost" /> instance.</param>
+    /// <param name="handler">The inner <see cref="T:System.Net.Http.HttpClientHandler" />.</param>
+    /// <param name="addressFilter">Selects what address should be used. If null, takes the first available address.</param>
+    /// <returns>An <see cref="T:System.Net.Http.HttpClient" /> configured to call the hosted application.</returns>
+    public static HttpClient CreateClient(this IHost host, HttpMessageHandler? handler = null, Func<Uri, bool>? addressFilter = null);
```
[the remark this answers is not in the document]
Comment (rafal-mz) [reply API/WithDocs/Microsoft.AspNetCore.Testing/ServiceFakesExtensions.cs:49]: - Create remarks section that explains it throws when there isn't IServer registered.

[the remark this answers is not in the document]
Comment (RussKie) [reply API/WithDocs/Microsoft.AspNetCore.Testing/ServiceFakesExtensions.cs:49]: > * Create remarks section that explains it throws when there isn't IServer registered.

No, that's what [`<exception />`](https://learn.microsoft.com/dotnet/csharp/language-reference/xmldoc/recommended-tags#exception) tag is for.

**Pull request closed without merging**